scripts/validate_docstrings.py

```diff
@@ -125,18 +169,22 @@ def needs_summary(self):
 
     @property
     def doc_parameters(self):
-        return self.doc['Parameters']
+        return {name: (type_, ''.join(desc))
+                for name, type_, desc in self.doc['Parameters']}
 
     @property
     def signature_parameters(self):
-        if not (inspect.isfunction(self.method_obj)
-                or inspect.isclass(self.method_obj)):
+        try:
+            signature = inspect.signature(self.method_obj)
+        except ValueError:
```

Comment:
this breaks for attributes (they raise a TypeError). Also, the test for {'dt', 'str', 'cat'} might need to be done before this one

Reply:
Ok, will take a look, I guess the solution is to also capture `TypeError`. With the `isfunc` and `isclass` we had before was raising an exception (I assume to everyone) in `Index.at` I think.

Reply:
Do you mean Series.at instead of Index.at? 
For `Series.at` I now get an error, while previously it worked.

Reply:
This is the error:
```
Traceback (most recent call last):
  File "./validate_docstrings.py", line 372, in <module>
    sys.exit(main(args.function))
  File "./validate_docstrings.py", line 356, in main
    return validate_all()
  File "./validate_docstrings.py", line 273, in validate_all
    int(doc.correct_parameters),
  File "./validate_docstrings.py", line 169, in correct_parameters
    return not bool(self.parameter_mismatches)
  File "./validate_docstrings.py", line 147, in parameter_mismatches
    signature_params = self.signature_parameters
  File "./validate_docstrings.py", line 139, in signature_parameters
    params = tuple(inspect.signature(self.method_obj).parameters.keys())
  File "/home/mgarcia/.anaconda3/envs/pandas_dev/lib/python3.6/inspect.py", line 3036, in signature
    return Signature.from_callable(obj, follow_wrapped=follow_wrapped)
  File "/home/mgarcia/.anaconda3/envs/pandas_dev/lib/python3.6/inspect.py", line 2786, in from_callable
    follow_wrapper_chains=follow_wrapped)
  File "/home/mgarcia/.anaconda3/envs/pandas_dev/lib/python3.6/inspect.py", line 2341, in _signature_from_callable
    'no signature found for builtin type {!r}'.format(obj))
ValueError: no signature found for builtin type <class 'pandas.core.indexing._AtIndexer'>
```